### PR TITLE
[Storage] Migrate first two test files to test proxy

### PR DESCRIPTION
**Pull request by jalauzon-msft**
This change adds the remaining infra code to use test proxy in Storage and migrates the first two files to test proxy (one sync and one async).

We are following a checklist of some additional things to cleanup all of our tests along the way as well. The checklist can be seen in #24936

### CODE REVIEW

**Comment by scbedd** [comment]
🚀 🚀 🚀

File: sdk/storage/azure-storage-blob/tests/test_append_blob.py [hunk 1]
```diff
@@ -1326,14 +1517,24 @@ def test_copy_sealed_blob_with_seal_blob_will_get_a_sealed_blob(self, storage_ac
...
+
+        variables = kwargs.pop("variables")
+        if self.is_live:
+            expiry_time = datetime.utcnow() + timedelta(seconds=10)
+            variables = {"expiry_time": expiry_time.isoformat()}
+
         bsc = BlobServiceClient(self.account_url(versioned_storage_account_name, "blob"), versioned_storage_account_key, max_block_size=4 * 1024)
         self._setup(bsc)
-
+
```
Comment (vincenttran-msft) [inline sdk/storage/azure-storage-blob/tests/test_append_blob.py:1537]: Is this a tab? I'm surprised PyLint hasn't kicked you to the curb for these ones, as compared to something like line 1509 which is just a newline

**Review by vincenttran-msft — APPROVED** [review]
Great work on this! Thanks for biting the bullet and doing all the hard stuff 😇 
I'm sure I'll be referencing this PR when I start working on my own test proxy migration :)

**Review by mccoyp — APPROVED** [review]

File: sdk/storage/azure-storage-blob/tests/test_append_blob.py (hunk 1 above)
Comment (jalauzon-msft) [reply sdk/storage/azure-storage-blob/tests/test_append_blob.py:1537]: It is a tab, strange. Yeah also surprised PyLint didn't yell. Will fix them up.

**Pull request merged**